contracts/PosterFactory.sol

```diff
@@ -0,0 +1,96 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/utils/CountersUpgradeable.sol";
+import "./metadata/ERC1155OnChainMetadata.sol";
+
+contract PosterFactory is ERC1155OnChainMetadata {
+    using CountersUpgradeable for CountersUpgradeable.Counter;
+
+    // Index of current PosterID.
+    CountersUpgradeable.Counter private tokenId;
+
+    function initialize(
+        string memory name_,
+        string memory symbol_,
+        string memory contract_description_,
+        string memory contract_image_,
+        string memory contract_external_link_,
+        uint256 contract_seller_fee_basis_points_,
+        address royaltyRecipient_
+    ) external initializer {
+        __ERC1155_init("");
+        __Context_init();
+        __ERC165_init();
+        name = name_;
+        symbol = symbol_;
+        contract_description = contract_description_;
+        contract_image = contract_image_;
+        contract_external_link = contract_external_link_;
+        contract_seller_fee_basis_points = contract_seller_fee_basis_points_;
```

Comment:
Let's add eip2981 to use `contract_seller_fee_basis_points` and `contract_fee_recipient`

Reply:
added ✅ 

<img width="830" alt="Screen Shot 2022-07-07 at 1 28 24 PM" src="https://user-images.githubusercontent.com/23249402/177824033-4b8d94df-e09b-4256-bfbd-991b81bc4409.png">